File: tools/FixLlmConfig.java
import java.sql.*;

public class FixLlmConfig {
    public static void main(String[] args) {
        String url = "jdbc:mysql://localhost:3306/smart_query";
        String user = "root";
        String password = "900110";

        try {
            // 加载MySQL驱动
            Class.forName("com.mysql.cj.jdbc.Driver");

            System.out.println("连接数据库...");
            Connection conn = DriverManager.getConnection(url, user, password);

            System.out.println("当前LLM配置:");
            System.out.println("====================");
            Statement stmt = conn.createStatement();
            ResultSet rs = stmt.executeQuery(
                "SELECT model_code, api_url, LEFT(api_key, 15) as key_preview, status " +
                "FROM sq_llm_config WHERE deleted=0 AND status=1"
            );

            while (rs.next()) {
                System.out.println("模型: " + rs.getString("model_code"));
                System.out.println("  URL: " + rs.getString("api_url"));
                System.out.println("  Key: " + rs.getString("key_preview") + "...");
                System.out.println("  状态: " + rs.getInt("status"));
                System.out.println();
            }

            System.out.println("修复API URL...");
            System.out.println("====================");
            int updated = stmt.executeUpdate(
                "UPDATE sq_llm_config " +
                "SET api_url = 'https://open.bigmodel.cn/api/coding/paas/v4/chat/completions', updated_at = NOW() " +
                "WHERE model_code IN ('glm-4', 'glm-5.1') AND deleted=0"
            );
            System.out.println("更新了 " + updated + " 条记录");

            System.out.println("\n修复后的配置:");
            System.out.println("====================");
            rs = stmt.executeQuery(
                "SELECT model_code, api_url, status " +
                "FROM sq_llm_config WHERE model_code IN ('glm-4', 'glm-5.1') AND deleted=0"
            );

            while (rs.next()) {
                System.out.println("模型: " + rs.getString("model_code"));
                System.out.println("  URL: " + rs.getString("api_url"));
                System.out.println("  状态: " + rs.getInt("status"));
                System.out.println();
            }

            rs.close();
            stmt.close();
            conn.close();

            System.out.println("✅ 配置修复完成！");

        } catch (Exception e) {
            System.err.println("错误: " + e.getMessage());
            e.printStackTrace();
        }
    }
}

File: tools/fix_llm_config.py
#!/usr/bin/env python3
"""
修复数据库中的LLM API URL配置
"""
import os
import subprocess
import sys

def fix_llm_config():
    """执行SQL修复数据库配置"""

    sql_commands = [
        # 查看当前配置
        "SELECT model_code, api_url, LEFT(api_key, 15) as key_preview FROM sq_llm_config WHERE deleted=0 AND status=1;",
        # 修复API URL
        "UPDATE sq_llm_config SET api_url = 'https://open.bigmodel.cn/api/coding/paas/v4/chat/completions', updated_at = NOW() WHERE model_code IN ('glm-4', 'glm-5.1') AND deleted=0;",
        # 验证修复结果
        "SELECT model_code, api_url, status FROM sq_llm_config WHERE model_code IN ('glm-4', 'glm-5.1') AND deleted=0;"
    ]

    print("正在修复LLM配置...")
    print("=" * 60)

    for i, sql in enumerate(sql_commands, 1):
        print(f"\n步骤 {i}: {sql[:50]}...")
        try:
            # 使用Windows的mysql命令
            result = subprocess.run(
                ['mysql', '-u', 'root', '-p900110', 'smart_query', '-e', sql],
                capture_output=True,
                text=True,
                encoding='gbk',  # Windows默认编码
                timeout=10
            )

            if result.returncode == 0:
                print(result.stdout)
            else:
                print(f"错误: {result.stderr}")

        except subprocess.CalledProcessError as e:
            print(f"执行失败: {e}")
        except Exception as e:
            print(f"异常: {e}")

    print("\n" + "=" * 60)
    print("配置修复完成！")

if __name__ == "__main__":
    try:
        fix_llm_config()
    except KeyboardInterrupt:
        print("\n操作已取消")
        sys.exit(1)
